drivers/hsc/hsc.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_hsc
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Honeywell HSC temperature and pressure sensor.
+ *
+ * @author      Quang Pham <phhr_quang@live.com>
+ *
+ * @}
+ */
+
+#include "math.h"
+#include "hsc.h"
+#include "hsc_internals.h"
+#include "hsc_params.h"
+#include "xtimer.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define DEV_I2C      (dev->params.i2c_dev)
+#define DEV_ADDR     (dev->params.i2c_addr)
+
+/* Internal function prototypes */
+static int _read_ut(const hsc_t *dev, int32_t *ut);
+static int _read_up(const hsc_t *dev, int32_t *up);
+
+/*---------------------------------------------------------------------------*
+ *                          HSC Core API                                     *
+ *---------------------------------------------------------------------------*/
+
+int hsc_init(hsc_t *dev, const hsc_params_t *params)
+{
+    dev->params = *params;
+    uint8_t buf[HSC_FULL_DATA_LENGTH] = {0};
+    
+    /* Acquire exclusive access */
+    i2c_acquire(DEV_I2C);
+    
+    if (i2c_read_bytes(DEV_I2C, DEV_ADDR, &buf[0], HSC_FULL_DATA_LENGTH, 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[hsc] read_bytes fails\n");
+
+        return -1;
+    }
+
+    i2c_release(DEV_I2C);
+    
+    /* Check the status of the sensor */
+    uint8_t status = buf[0];
+
+    switch (status & HSC_STATUS_MASK) {
+        case HSC_STATUS_OK:
+            return  0;
+        case HSC_STATUS_STALE_DATA:
+            return -1;
+        default:
+            return -2;
+    }
+
+    return 0;
+}
+
+int16_t hsc_read_temperature(const hsc_t *dev)
+{
+    int32_t ut = 0;
+
+    /* Read uncompensated temperature value */
+    _read_ut(dev, &ut);
+
+    /* Compute true temperature value following datasheet formulas */
+    return (int16_t)(ut * 200 / 2047 - 50);
+}
+
+int32_t hsc_read_pressure(const hsc_t *dev)
+{
+    int32_t up = 0;
+
+    /* Read uncompensated pressure value */
+    _read_up(dev, &up);
+
+    /* Compute true pressure value following datasheet formulas */
+    return (int32_t)((up - 1638) * 2000 * HSC_RANGE / (14745 - 1638) - 1000 * HSC_RANGE);
+}
+
+/*------------------------------------------------------------------------------------*/
+/*                                Internal functions                                  */
+/*------------------------------------------------------------------------------------*/
+
+static int _read_ut(const hsc_t *dev, int32_t *output)
+{
+    /* Read UT (Uncompsensated Temperature value) */
+    uint8_t buf[HSC_FULL_DATA_LENGTH] = {0};
```

Comment:
```suggestion
    uint8_t buf[HSC_FULL_DATA_LENGTH];
```

no need to set this to 0, will be overwritten anyway.